UnitTests/SceneTests/SceneLoader.cpp

```diff
@@ -0,0 +1,261 @@
+#include <gtest/gtest.h>
+#include <iostream>
+#include <string>
+#include <fcntl.h>
+#include <stdlib.h>
+
+extern "C"
+{
+	#include "../../headers/mini_rt.h"
+	t_scene scene; 
+}
+
+void init_scene(t_scene *scene);
+void init_array(double *array, double x, double y, double z);
+void point_cmp(t_point *point1, double values[3]);
+void check_ambient_light(t_a_light *amb_light, double ratio, int color);
+void check_camera(t_camera *camera, double origin[3], double orientation[3], int fov);
+void check_light(t_light *light, double origin[3], double b_ratio);
+void check_plane(union u_object *u, double origin[3], double normal[3], int color);
+void check_sphere(union u_object *u, double origin[3], double sp_diam, int color);
+void check_cylinder(union u_object *u, double origin[3], double orientation[3], \
+	double diam1, double diam2, int color);
+
+TEST(SceneTests, SceneLoaderValid)
+{
+	t_scene scene;
+	double	origin[3];
+	double	orientation[3];
+	int		scene_fd;
+
+	{
+		scene_fd = open("../../../scenes/scene.rt", O_RDONLY);
+		EXPECT_NE(scene_fd, -1);
```

Comment:
Maybe it would be better to use ASSERT_NE here since the tests can't go on when the file fails to open.

Reply:
that`s a good point, didn`t think about that. I will change the code to use ASSERT_NE in cases like this.